File: Final/erp-frontend/erp-frontend/src/app/services/cliente.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ResponseWrapper, Cliente } from '../models/response-wrapper.model';

@Injectable({
  providedIn: 'root'
})
export class ClienteService {
  private baseUrl = 'http://localhost:8080/api/clientes';

  constructor(private http: HttpClient) { }

  crear(cliente: Cliente): Observable<ResponseWrapper<Cliente>> {
    return this.http.post<ResponseWrapper<Cliente>>(this.baseUrl, cliente);
  }

  listar(): Observable<ResponseWrapper<Cliente[]>> {
    return this.http.get<ResponseWrapper<Cliente[]>>(this.baseUrl);
  }

  obtenerPorId(id: number): Observable<ResponseWrapper<Cliente>> {
    return this.http.get<ResponseWrapper<Cliente>>(`${this.baseUrl}/${id}`);
  }

  actualizar(id: number, cliente: Cliente): Observable<ResponseWrapper<Cliente>> {
    return this.http.put<ResponseWrapper<Cliente>>(`${this.baseUrl}/${id}`, cliente);
  }

  desactivar(id: number): Observable<ResponseWrapper<void>> {
    return this.http.delete<ResponseWrapper<void>>(`${this.baseUrl}/${id}`);
  }

  activar(id: number): Observable<ResponseWrapper<void>> {
    return this.http.put<ResponseWrapper<void>>(`${this.baseUrl}/${id}/activar`, {});
  }

  buscarPorNombre(nombre: string): Observable<ResponseWrapper<Cliente[]>> {
    return this.http.get<ResponseWrapper<Cliente[]>>(`${this.baseUrl}/buscar?nombre=${nombre}`);
  }
}

File: Final/erp-backend/src/main/java/com/pyme/erp/service/ClienteService.java
package com.pyme.erp.service;

import com.pyme.erp.dto.request.ClienteRequest;
import com.pyme.erp.model.Cliente;

import java.util.List;

public interface ClienteService {
    
    Cliente crear(ClienteRequest request);
    
    Cliente actualizar(Long id, ClienteRequest request);
    
    Cliente obtenerPorId(Long id);
    
    List<Cliente> listarActivos();
    
    void desactivar(Long id);
    
    void activar(Long id);
    
    List<Cliente> buscarPorNombre(String nombre);
}